Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/random_perturbation.h

```diff
@@ -0,0 +1,233 @@
+  typename GT::Vector_3 construct_random_vector_3(const double& max_size,
+                                                  RNG& rng,
+                                                  const GT& gt)
+  {
+    typedef typename GT::FT FT;
+    typename GT::Construct_vector_3 vec = gt.construct_vector_3_object();
+
+    return vec(FT(rng.get_double(-max_size, max_size)),
+               FT(rng.get_double(-max_size, max_size)),
+               FT(rng.get_double(-max_size, max_size)));
+  }
+
+  template<typename GT, typename VertexRange,
+           typename PM, typename VCMap, typename VPMap, typename RNG>
+  void random_perturbation_impl(VertexRange vrange,
+                                PM& tmesh,
+                                const double& max_size,
+                                VCMap vcmap,
+                                VPMap vpmap,
+                                bool do_project,
+                                RNG& rng,
+                                const GT& gt)
+  {
+    typedef typename boost::graph_traits<PM>::vertex_descriptor vertex_descriptor;
+    typedef typename GT::Point_3    Point_3;
+    typedef typename GT::FT         FT;
+
+    typedef CGAL::AABB_face_graph_triangle_primitive<PM> Primitive;
+    typedef CGAL::AABB_traits<GT, Primitive> Traits;
+    typedef CGAL::AABB_tree<Traits> Tree;
+
+    Tree tree;
+    if(do_project)
+    {
+      tree.rebuild(faces(tmesh).first, faces(tmesh).second, tmesh);
+      tree.accelerate_distance_queries();
+    }
+    typename GT::Construct_translated_point_3 translate
+      = gt.construct_translated_point_3_object();
+
+    BOOST_FOREACH(vertex_descriptor v, vrange)
+    {
+      if (!get(vcmap, v) && !is_border(v, tmesh))
+      {
+        const Point_3& p = get(vpmap, v);
+        const Point_3 np = translate(p, construct_random_vector_3<GT>(max_size, rng, gt));
+
+        if (do_project)
+          put(vpmap, v, tree.closest_point(np)); //project on input surface
+        else
+          put(vpmap, v, np);
+      }
+    }
+  }
+
+} //end namespace internal
+
+/*!
+* \ingroup PMP_meshing_grp
+* @brief randomly perturbs the locations of vertices of a triangulated surface mesh.
+* In the absence of instructions to the contrary in the parameters,
+* the vertices are re-projected to the input surface after vertices geometric perturbation.
+* Note that depending on the chosen parameters, inversions and self-intersections may happen.
+*
+* @tparam VertexRange model of `Range`, holding
+*         vertices of type `boost::graph_traits<TriangleMesh>::vertex_descriptor`.
+*         Its iterator type is `ForwardIterator`. 
+* @tparam TriangleMesh model of `MutableFaceGraph`.
+* @tparam NamedParameters a sequence of \ref namedparameters
+*
+* @param vertices the range of vertices to be perturbed
+* @param tmesh the triangulated surface mesh to which the perturbed vertices belong
+* @param perturbation_max_size the maximal length of moves that can be applied to
```

Comment:
I guess it does not take into account the reprojection, right? If so we should mention it.